### PR TITLE
fix: Check empty value when converting Model to UpdateDTO

### LINKED ISSUE #528: [V2] Check empty value when converting Model to UpdateDTO
1. Check the empty value when convert Model to UpdateDTO
      - FromProvisionWatcherModelToUpdateDTO
      - FromDeviceModelToUpdateDTO
      - FromDeviceServiceModelToUpdateDTO

2. API client lib handle the 404 StatusNotFound response
3. Modify BinaryReading validate annotation

### PR DESCRIPTION
**Pull request by weichou1229**
Fix #528

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
1. Check the empty value when convert Model to UpdateDTO
      - FromProvisionWatcherModelToUpdateDTO
      - FromDeviceModelToUpdateDTO
      - FromDeviceServiceModelToUpdateDTO

2. API client lib handle the 404 StatusNotFound response
3. Modify BinaryReading validate annotation

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: v2/clients/http/utils/common.go [hunk 1]
```diff
@@ -150,6 +150,10 @@ func sendRequest(ctx context.Context, req *http.Request) ([]byte, errors.EdgeX)
 	}
 
 	// Handle error response
+	if resp.StatusCode == http.StatusNotFound {
```
Comment (cloudxxx8) [inline v2/clients/http/utils/common.go:153]: the condition should be `resp.StatusCode >= http.StatusBadRequest`, because they are all error status

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: v2/clients/http/utils/common.go (hunk 1 above)
Comment (weichou1229) [reply v2/clients/http/utils/common.go]: Modified.
Return error when resp.StatusCode > http.StatusMultiStatus:
```
	if resp.StatusCode <= http.StatusMultiStatus {
		return bodyBytes, nil
	}

	// Handle error response
	msg := fmt.Sprintf("request failed, status code: %d, err: %s", resp.StatusCode, string(bodyBytes))
	errKind := errors.KindMapping(resp.StatusCode)
	return nil, errors.NewCommonEdgeX(errKind, msg, nil)
```

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by lenny-intel** [comment]
recheck

**Pull request merged**